packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js

```diff
@@ -35,34 +40,37 @@ function TaxonomiesPanel(props) {
     return null;
   }
 
-  // TODO: remove this eventually
-  // show categories before tags
-  const sortedTaxonomies = taxonomies.sort(
-    ({ restBase: restBaseA }, { restBase: restBaseB }) => {
-      if (restBaseA > restBaseB) {
-        return 1;
-      } else if (restBaseB > restBaseA) {
-        return -1;
-      }
-      return 0;
-    }
-  );
+  const availableTaxonomies = taxonomies.filter((taxonomy) => {
+    const isVisible = taxonomy?.visibility?.show_ui;
+    const canAssignTerms = Boolean(capabilities[`assign-${taxonomy?.slug}`]);
+
+    return isVisible && canAssignTerms && taxonomy;
```

Comment:
what's the need for the `&& taxonomy` here? isn't that always gonna be an object?

Reply:
ah, yup!